services/wfm/src/main/java/org/openkilda/wfm/share/mappers/FlowMapper.java

```diff
@@ -0,0 +1,126 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.share.mappers;
+
+import org.openkilda.messaging.model.Flow;
+import org.openkilda.messaging.payload.flow.FlowState;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+
+import java.time.Instant;
+
+/**
+ * Convert {@link org.openkilda.model.Flow} to {@link Flow} and back.
+ */
+@Mapper
+public abstract class FlowMapper {
+
+    public static final FlowMapper INSTANCE = Mappers.getMapper(FlowMapper.class);
+
+    @Mapping(source = "srcSwitchId", target = "sourceSwitch")
+    @Mapping(source = "destSwitchId", target = "destinationSwitch")
+    @Mapping(source = "status", target = "state")
+    public abstract Flow map(org.openkilda.model.Flow flow);
+
+    @Mapping(source = "sourceSwitch", target = "srcSwitchId")
+    @Mapping(source = "destinationSwitch", target = "destSwitchId")
+    @Mapping(source = "state", target = "status")
+    public abstract org.openkilda.model.Flow map(Flow flow);
+
+    /**
+     * Convert {@link org.openkilda.model.SwitchId} to {@link org.openkilda.messaging.model.SwitchId}.
+     */
+    public org.openkilda.messaging.model.SwitchId map(org.openkilda.model.SwitchId value) {
```

Comment:
Why for we have a copy of SwitchId class into messaging?

Reply:
Fixed.